pkg/client/testclient/fixture.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testclient
+
+import (
+	"fmt"
+	"io/ioutil"
+	"reflect"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/meta"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/yaml"
+)
+
+type ObjectRetriever interface {
+	Kind(kind, name string) (runtime.Object, error)
+	Add(runtime.Object) error
+}
+
+func ObjectReaction(o ObjectRetriever, mapper meta.RESTMapper) ReactionFunc {
+	return func(action FakeAction) (runtime.Object, error) {
+		segments := strings.Split(action.Action, "-")
+		var verb, resource string
+		switch len(segments) {
+		case 3:
+			verb, _, resource = segments[0], segments[1], segments[2]
+		case 2:
+			verb, resource = segments[0], segments[1]
+		default:
+			return nil, fmt.Errorf("unrecognized action, need two or three segments <verb>-<resource> or <verb>-<subresource>-<resource>: %s", action.Action)
+		}
+		_, kind, err := mapper.VersionAndKindForResource(resource)
+		if err != nil {
+			return nil, fmt.Errorf("unrecognized action %s: %v", resource, err)
+		}
+		// TODO: have mapper return a Kind for a subresource?
+		switch verb {
+		case "list", "search":
+			return o.Kind(kind+"List", "")
+		case "get", "create", "update", "delete":
+			// TODO: handle sub resources
+			if s, ok := action.Value.(string); ok && action.Value != nil {
+				return o.Kind(kind, s)
+			}
+			return o.Kind(kind, "unknown")
+		default:
+			return nil, fmt.Errorf("no reaction implemented for %s", action.Action)
+		}
+		return nil, nil
+	}
+}
+
+type objects struct {
+	types   map[string][]runtime.Object
+	last    map[string]int
+	typer   runtime.ObjectTyper
+	creater runtime.ObjectCreater
+	copier  copier
+}
+
+type copier interface {
+	Copy(obj runtime.Object) (runtime.Object, error)
+}
+
+func NewObjects(scheme *runtime.Scheme) ObjectRetriever {
+	return objects{
+		types:   make(map[string][]runtime.Object),
+		last:    make(map[string]int),
+		typer:   scheme,
+		creater: scheme,
+		copier:  scheme,
+	}
+}
+
+func AddObjectsFromPath(path string, o ObjectRetriever) error {
```

Comment:
I think it will be good to add a comment saying that this path is a path to a yaml file which has the objects that we want to add to the object retriever.